packages/app-store/salesforce/lib/CrmService.ts

```diff
@@ -497,32 +477,42 @@ export default class SalesforceCRMService implements CRM {
         appOptions.createEventOnLeadCheckForContact &&
         !forRoundRobinSkip
       ) {
-        // Get any matching contacts
-        const contactSearch = await conn.query(
-          `SELECT Id, Email, OwnerId, Owner.Email FROM ${
-            SalesforceRecordEnum.CONTACT
-          } WHERE Email IN ('${emailArray.join("','")}')`
-        );
-
-        if (contactSearch?.records?.length > 0) {
-          records = contactSearch.records as ContactRecord[];
-          this.setFallbackToContact(true);
-          log.info(
-            "Found matching contacts, falling back to contact",
-            safeStringify({
-              contactCount: records.length,
-            })
-          );
+        const record = await this.getContactOrLeadFromEmail({
+          email: emailArray[0],
+          setFallbackToContact: true,
+          conn,
+        });
+        if (record) {
+          records.push(record);
         }
       }
 
-      if (!records.length && results?.records?.length) {
-        records = results.records as ContactRecord[];
-      }
+      if (records.length === 0) {
+        // Handle Account record type
+        if (recordToSearch === SalesforceRecordEnum.ACCOUNT) {
```

Comment:
If only searching for a single record then we make this single call if the above two cases aren't true.